contracts/src/types/V2.sol

```diff
@@ -0,0 +1,93 @@
+// SPDX-License-Identifier: Apache-2.0
+// SPDX-FileCopyrightText: 2023 Snowfork <hello@snowfork.com>
+pragma solidity 0.8.25;
+
+import {ParaID} from "./Common.sol";
+
+struct Origin {
+    // Origin network
+    uint8 network;
+    // Origin chain
+    ParaID paraID;
+    // Stable hash of full origin location;
+    bytes32 locationID;
+}
+
+// Inbound message from a Polkadot parachain (via BridgeHub)
+struct InboundMessage {
+    // origin
+    bytes32 origin;
+    // Message nonce
+    uint64 nonce;
+    // Commands
+    Command[] commands;
+}
+
+struct Command {
+    uint8 kind;
+    bytes payload;
+}
```

Comment:
Seems we remove `maxDispatchGas` from the Command,  so we depend on some api service dryRun https://docs.metamask.io/wallet/reference/eth_estimategas/  for the gas cost?

Reply:
latest commit updates the Command to include a `gas` param.